src/components/card/index.tsx

```diff
@@ -1,13 +1,170 @@
 import * as React from 'react'
+import {createNamedContext} from '@reach/utils/context'
+import {useId} from '@reach/auto-id'
+import type * as Polymorphic from '@reach/utils/polymorphic'
+
+const CardContext = createNamedContext<InternalCardContextValue>(
+  'CardContext',
+  {} as InternalCardContextValue,
+)
+
+type InternalCardContextValue = {
+  cardId: string | undefined
+  quiet: boolean
+  horizontal: boolean
+}
+
+type CardProps = {
+  quiet?: boolean
+  horizontal?: boolean
+}
+
+const cardDefaultClasses = `bg-white dark:bg-gray-800 dark:text-gray-200 shadow-sm rounded-lg overflow-hidden sm:p-8 p-5`
+
+const Card = React.forwardRef(function Card(
+  {
+    children,
+    as: Comp = 'div',
+    quiet = false,
+    horizontal = false,
+    className,
+    ...props
+  },
+  forwardRef,
+) {
+  if (!quiet) {
+    className = `${cardDefaultClasses} ${className ? className : ''}`
+  }
+  const id = useId(props.id)
+  const context: InternalCardContextValue = React.useMemo(
+    () => ({
+      cardId: id,
+      quiet,
+      horizontal,
+    }),
+    [quiet, id, horizontal],
+  )
 
-const Card: React.FC = ({children}) => {
   return (
-    <div
-      className={`bg-white dark:bg-gray-800 dark:text-gray-200 shadow-sm rounded-lg overflow-hidden sm:p-5 p-4`}
-    >
+    <CardContext.Provider value={context}>
+      <Comp
+        className={className}
+        {...props}
+        ref={forwardRef}
+        data-egghead-card=""
```

Comment:
Reach does this and it allows you to target elements with global styles. I think this is an interesting approach!